pkg/client/client.go

```diff
@@ -49,18 +50,25 @@ func WithTLSConfig(tlsConfig *tls.Config) Option {
 	}
 }
 
-// Opts is used to interact with an EDV operation.
-type Opts struct {
-	httpHeader map[string]string
+// WithHeaders option is for setting additional http request headers (since it's a function, it can call a remote
+func WithHeaders(addHeadersFunc addHeaders) Option {
+	return func(opts *Client) {
+		opts.headersFunc = addHeadersFunc
+	}
 }
 
-// EDVOption configures the edv operation
-type EDVOption func(opts *Opts)
+// ReqOpts is used to interact with an EDV operation.
+type ReqOpts struct {
+	addHeadersFunc addHeaders
+}
 
-// WithHTTPHeader option is
-func WithHTTPHeader(key, value string) EDVOption {
-	return func(opts *Opts) {
-		opts.httpHeader[key] = value
+// ReqOption edv req option
+type ReqOption func(opts *ReqOpts)
+
+// WithRequestHeader option is for setting additional http request headers (since it's a function, it can call a remote
```

Comment:
Unclosed bracket here too. Is the comment cut off?

Reply:
done